js/src/logger.ts

```diff
@@ -2863,6 +3000,39 @@ function validateAndSanitizeExperimentLogPartialArgs(
   }
 }
 
+/**
+ * Helper function for uploading attachments. Recursively extracts `Attachment` values.
+ * @param event The event to be extract. Will be modified.
+ * @returns The function to get the modified event and the extracted `Attachment` objects.
+ */
+function extractAttachments<T extends Partial<BackgroundLogEvent>>(
+  event: T,
+): Attachment[] {
+  const attachments: Attachment[] = [];
+
+  function helper(value: any) {
+    if (typeof value !== "object") {
```

Comment:
Probably want to check `value instanceof Object` here, since `typeof null === "object"`

Reply:
Done